client/main.lua

```diff
@@ -1,65 +1,121 @@
+        local plant = propertyPlants[currentProperty][i]
+        local plantProp = CreateObject(joaat(sharedConfig.plants[plant.sort].stages[plant.stage]), plant.coords.x, plant.coords.y, plant.coords.z, false, false, false)
+        while not plantProp do
+            Wait(0)
+        end
 
         PlaceObjectOnGroundProperly(plantProp)
         Wait(10)
         FreezeEntityPosition(plantProp, true)
         SetEntityAsMissionEntity(plantProp, false, false)
     end
+
     plantsSpawned = true
 end
 
+local function spawnOutsidePlants()
+    if plantsSpawned or currentProperty then return end
+
+    for i = 1, #outsidePlants do
+        local plant = outsidePlants[i]
+        local plantProp = CreateObject(joaat(sharedConfig.plants[plant.sort].stages[plant.stage]), plant.coords.x, plant.coords.y, plant.coords.z, false, false, false)
+        while not plantProp do
+            Wait(0)
+        end
+
+        PlaceObjectOnGroundProperly(plantProp)
+        Wait(10)
+        FreezeEntityPosition(plantProp, true)
+        SetEntityAsMissionEntity(plantProp, false, false)
+    end
+
+    plantsSpawned = true
+end
+
+---@param plantStage 1 | 2 | 3 | 4 | 5 | 6 | 7
+---@param plantData WeedPlant
 local function deleteClosestPlant(plantStage, plantData)
-    if not (plantStage and plantData) then return end
+    if not plantStage or not plantData then return end
 
-    local closestPlant = GetClosestObjectOfType(plantData.coords.x, plantData.coords.y, plantData.coords.z, 3.5, joaat(plantStage), false, false, false)
+    local closestPlant = GetClosestObjectOfType(plantData.coords.x, plantData.coords.y, plantData.coords.z, 3.5, joaat(sharedConfig.plants[plantData.sort].stages[plantStage]), false, false, false)
     if closestPlant == 0 then return end
 
     DeleteObject(closestPlant)
 end
 
-local function despawnPlants()
-    if not (plantsSpawned or currentHouse) then return end
+---@param property string
+local function despawnPropertyPlants(property)
+    if not plantsSpawned or not property then return end
 
-    local plants = housePlants[currentHouse]
+    local plants = propertyPlants[property]
+    if not plants then return end
 
-    for i = #plants, 1, -1 do
+    for i = 1, #plants do
         local plantData = plants[i]
-        for _, stage in pairs(sharedConfig.plants[plantData.sort].stages) do
+        for stage = 1, #sharedConfig.plants[plantData.sort].stages do
+            deleteClosestPlant(stage, plantData)
+        end
+
+        plants[i] = nil
+    end
+
+    plantsSpawned = false
+end
+
+---@param property string
+local function despawnOutsidePlants(property)
+    if not plantsSpawned or property then return end
+
+    for i = 1, #outsidePlants do
+        local plantData = outsidePlants[i]
+        for stage = 1, #sharedConfig.plants[plantData.sort].stages do
             deleteClosestPlant(stage, plantData)
         end
-        table.remove(plants, i)
+
+        outsidePlants[i] = nil
```

Comment:
If each entry in the table is going to be set to nil, would it be easier to set the entire table to empty table '{}' at the conclusion of the for loop?

Reply:
It was doing the same before so I kept it the same but without it using table.remove as that is unnecessary.